src/main/java/org/killbill/billing/plugin/avatax/api/AvaTaxTaxCalculator.java

```diff
@@ -203,10 +207,22 @@ private CreateTransactionModel toTaxRequest(final String companyCode,
                                                 final boolean dryRun,
                                                 final boolean shouldCommitDocuments,
                                                 final Iterable<PluginProperty> pluginProperties,
-                                                final LocalDate utcToday) {
-        Preconditions.checkState((originalInvoiceReferenceCode == null && (adjustmentItems == null || adjustmentItems.isEmpty())) ||
-                                 (originalInvoiceReferenceCode != null && (adjustmentItems != null && !adjustmentItems.isEmpty())),
-                                 "Invalid combination of originalInvoiceReferenceCode %s and adjustments %s", originalInvoiceReferenceCode, adjustmentItems);
+                                                final LocalDate utcToday,
+                                                final boolean skipAnomalousAdjustments) {
+        
+        try {
+            Preconditions.checkState((originalInvoiceReferenceCode == null && (adjustmentItems == null || adjustmentItems.isEmpty())) ||
+                                        (originalInvoiceReferenceCode != null && (adjustmentItems != null && !adjustmentItems.isEmpty())),
+                                        "Invalid combination of originalInvoiceReferenceCode %s and adjustments %s", originalInvoiceReferenceCode, adjustmentItems);
+        } catch (IllegalStateException e) {                                     
+            if (skipAnomalousAdjustments) {
+                logger.warn("Returning null tax request instead of throwing IllegalStateException exception for Invalid combination of originalInvoiceReferenceCode {} and adjustments {}",
```

Comment:
The log message makes sense from a code perspective, but isn't really easy to understand when looking at it in tools like Splunk. What about something like:

```
logger.warn("Ignoring tax request due to inconsistent adjustments: originalInvoiceReferenceCode={}, adjustmentItems={}", originalInvoiceReferenceCode, adjustmentItems);
```

Reply:
this looks better, thanks for the suggestion. will update

Reply:
updated in latest commit